Asignment7/baseball_stats.sql

```diff
@@ -0,0 +1,485 @@
+        SELECT t1.team_id,
+               t1.game_id,
+               t1.local_date,
+               t1.pitcher,
+               (SUM(t2.Walk*9)/SUM(t2.innings_pitched))AS BB_per_9_innings
+            FROM s_starting_pitcher_table AS t1 JOIN s_starting_pitcher_table AS t2
+            ON t1.team_id= t2.team_id
+            WHERE t2.local_date >= DATE_ADD(t1.local_date, INTERVAL -200 DAY)
+            AND t2.local_date< t1.local_date
+         GROUP BY t1.team_id, t1.local_date,t1.game_id;
+
+-- calculating hits allowed per 9 innings pitched for starting pitcher
+
+DROP TABLE IF EXISTS s_starting_pitcher_H_per_9inn;
+CREATE TABLE s_starting_pitcher_H_per_9inn
+    (INDEX GAME_ID_INd(game_id),INDEX TEAM_ID_INd(team_id), INDEX H_per_9_innings_IND(H_per_9_innings))
+        SELECT t1.team_id,
+               t1.game_id,
+               t1.local_date,
+               t1.pitcher,
+               (SUM(t2.Hit*9)/SUM(t2.innings_pitched))AS H_per_9_innings
+            FROM s_starting_pitcher_table AS t1 JOIN s_starting_pitcher_table AS t2
+            ON t1.team_id= t2.team_id
+            WHERE t2.local_date >= DATE_ADD(t1.local_date, INTERVAL -200 DAY)
+            AND t2.local_date< t1.local_date
+         GROUP BY t1.team_id, t1.local_date,t1.game_id;
+
+-- calculating strikeouts per innings pitched for starting pitcher
+DROP TABLE IF EXISTS s_starting_pitcher_K_per_9inn;
+CREATE TABLE s_starting_pitcher_K_per_9inn
+    (INDEX GAME_ID_INd(game_id),INDEX TEAM_ID_INd(team_id), INDEX k_per_9_innings_IND(k_per_9_innings))
+        SELECT t1.team_id,
+               t1.game_id,
+               t1.local_date,
+               t1.pitcher,
+               (SUM(t2.StrikeOut * 9)/SUM(t2.innings_pitched))AS k_per_9_innings
+            FROM s_starting_pitcher_table AS t1 JOIN s_starting_pitcher_table AS t2
+            ON t1.team_id= t2.team_id
+            WHERE t2.local_date >= DATE_ADD(t1.local_date, INTERVAL -200 DAY)
+            AND t2.local_date< t1.local_date
+         GROUP BY t1.team_id, t1.local_date,t1.game_id;
+
+-- calculating rolling WHIP for starting pitcher
+
+DROP TABLE IF EXISTS s_starting_pitcher_rolling_WHIP;
+CREATE TABLE s_starting_pitcher_rolling_WHIP
+    (INDEX GAME_ID_INd(game_id),INDEX TEAM_ID_INd(team_id), INDEX starting_pitcher_rolling_WHIP_IND(starting_pitcher_rolling_WHIP))
+        SELECT t1.team_id,
+               t1.game_id,
+               t1.local_date,
+               t1.pitcher,
+               IF (SUM(t2.innings_pitched)=0,0,(SUM(t2.Hit) + SUM(t2.Walk))/(SUM(t2.innings_pitched))) AS starting_pitcher_rolling_WHIP
+            FROM s_starting_pitcher_table AS t1 JOIN s_starting_pitcher_table AS t2
+            ON t1.team_id= t2.team_id
+            WHERE t2.local_date >= DATE_ADD(t1.local_date, INTERVAL -200 DAY)
+            AND t2.local_date< t1.local_date
+         GROUP BY t1.team_id, t1.local_date,t1.game_id;
+
+-- appending game and boxscore table
+
+DROP TABLE IF EXISTS s_baseball_game_stats;
```

Comment:
If you're taking a response from winner_home_or_away from the boxscore table, you might want to cross-check the results as I realized that values in that column are not always correct. Take a look:
`SELECT game_id, home_runs, away_runs, winner_home_or_away
FROM boxscore
WHERE game_id IN (333417, 325875)
ORDER BY game_id;`